core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1142,34 +1176,29 @@ private boolean stateNotActive() {
         return  partitionState() != SharePartitionState.ACTIVE;
     }
 
-    private void completeInitializationWithException(CompletableFuture<Void> future, Throwable exception) {
+    private void completeInitializationWithException() {
         lock.writeLock().lock();
         try {
             partitionState = SharePartitionState.FAILED;
         } finally {
             lock.writeLock().unlock();
         }
-        future.completeExceptionally(exception);
     }
 
-    private void maybeCompleteInitialization(CompletableFuture<Void> future) {
+    private InitializationResult maybeCompleteInitialization() {
```

Comment:
I think I'd call it `checkInitializationCompletion`. This method does not complete initialization at all. It tell you whether it's complete or not.

Reply:
done